tests/utils/serializeDate.test.ts

```diff
@@ -0,0 +1,16 @@
+import { serializeDate } from '../../src/utils/serializeDate';
+
+describe('serializeDate', () => {
+  it.each`
+    type           | value
+    ${'string'}    | ${'test123'}
+    ${'undefined'} | ${undefined}
+  `('passes $type parameter through untouched', ({ value }) => {
+    expect(serializeDate(value)).toBe(value);
+  });
+
+  it('returns ISO date string when the parameter is a Date object', () => {
```

Comment:
may be also add a test to check if the input is not an instance of Date?

Reply:
Good point @vishalbrlife. I overlooked the fact that we won't be able to check for the parameter type when we compile down to JavaScript.

Currently, I'm unable to pass in a parameter outside of the defined types. Let's address this issue in another PR so that we don't further increase the size of this PR. What do you think?

Reply:
Added in ae62e0d